### DISCUSSION TITLE
Way to get molecules imperatively

### DISCUSSION THREAD

**Discussion by alexlag** [saasquatch/bunshi#55] [Q&A] [answered]
I am looking for a way to get molecule's actual state from imperative callbacks similar to jotai's `useAtomCallback`.

My experiments lead me to following snippet:
```typescript
const injector = useInjector();
const scopes = useScopes();

const getMolecule = useCallback(
  <T>(mol: MoleculeOrInterface<T>) => injector.get(mol, ...scopes),
  [injector, scopes],
);
```
and passing `getMolecule` around. 

Is this a correct way (assuming mostly implicit scopes inside relative component)?

**Comment by loganvolkers** [comment, marked as answer]
That will work, but keep in mind that calling `get` on the injector is not the same as calling `use`.

Using `useMolecule` is still recommended for most cases since it handles the [subscription lifecycle](https://www.bunshi.org/concepts/lifecycle/) for you automatically.
